--- a/usr/sbin/pocketchip-warn05
+++ b/usr/sbin/pocketchip-warn05
@@ -1,7 +1,7 @@
 #!/bin/bash
 VOLT=""
 
-if [ ! -f $HOME/.pocketchip-batt/warn05 ] && [ $(cat /usr/lib/pocketchip-batt/charging) = "0" ]; then
+if [ ! -f $HOME/.pocketchip-batt/warn05 ] && [ "$(cat /usr/lib/pocketchip-batt/charging)" = "0" ]; then
 
   while [ -z "$VOLT" ]
   do
@@ -13,7 +13,7 @@
     mkdir -p $HOME/.pocketchip-batt
     touch $HOME/.pocketchip-batt/warn05
   elif [ $(echo " $VOLT < 3400" | bc ) -eq 1 ]; then
-    rm $HOME/.pocketchip-batt/warn05
+    rm -f $HOME/.pocketchip-batt/warn05
   fi
 
 #elif [ -f $HOME/.pocketchip-batt/warn05 ] && [ $(cat /usr/lib/pocketchip-batt/charging) = "1" ]; then
@@ -25,7 +25,7 @@
   done
 
   if [ $(echo " $VOLT > 3400" | bc ) -eq 1 ]; then
-    rm $HOME/.pocketchip-batt/warn05
+    rm -f $HOME/.pocketchip-batt/warn05
   fi
 
 
--- a/usr/sbin/pocketchip-warn15
+++ b/usr/sbin/pocketchip-warn15
@@ -1,7 +1,7 @@
 #!/bin/bash
 VOLT=""
 
-if [ ! -f $HOME/.pocketchip-batt/warn15 ] && [ $(cat /usr/lib/pocketchip-batt/charging) = "0" ]; then
+if [ ! -f $HOME/.pocketchip-batt/warn15 ] && [ "$(cat /usr/lib/pocketchip-batt/charging)" = "0" ]; then
 
   while [ -z "$VOLT" ]
   do
@@ -13,7 +13,7 @@
     mkdir -p $HOME/.pocketchip-batt
     touch $HOME/.pocketchip-batt/warn15
   elif [ $(echo " $VOLT > 3550" | bc) -eq 1 ]; then
-  	rm $HOME/.pocketchip-batt/warn*
+  	rm -f $HOME/.pocketchip-batt/warn*
   fi
 
 #elif [ -f $HOME/.pocketchip-batt/warn15 ] && [ $(cat /usr/lib/pocketchip-batt/charging) = "1" ]; then
@@ -25,7 +25,7 @@
   done
 
   if [ $(echo " $VOLT > 3550" | bc ) -eq 1 ]; then
-    rm $HOME/.pocketchip-batt/warn*
+    rm -f $HOME/.pocketchip-batt/warn*
   fi
 
 fi
